cmd/allocator/main.go

```diff
@@ -64,6 +63,18 @@ const (
 	sslPort = "8443"
 )
 
+// grpcHandlerFunc returns an http.Handler that delegates to grpcServer on incoming gRPC
+// connections or otherHandler otherwise. Copied from https://github.com/philips/grpc-gateway-example.
+func grpcHandlerFunc(grpcServer http.Handler, otherHandler http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		// This is a partial recreation of gRPC's internal checks https://github.com/grpc/grpc-go/pull/514/files#diff-95e9a25b738459a2d3030e1e6fa2a718R61
+		if r.ProtoMajor == 2 && strings.Contains(r.Header.Get("Content-Type"), "application/grpc") {
```

Comment:
I am still not clear why `ProtoMajor == 2` and not 3 for the gRPC server.
Can you please add more documentations?

Reply:
`ProtoMajor` refers to the HTTP protocol major version part. We are checking if request is HTTP/1.1 or HTTP/2. Will add that as a comment to the code.